### PR TITLE
Install zfs

### PR DESCRIPTION
**Pull request by binocarlos**
This PR checks to see if the `c.config["agent_config"]["dataset"]["backend"]` is set to `zfs` and if yes will install the ZFS packages and mount a flocker pool ready for use by the ZFS dataset agent

### CODE REVIEW

**Comment by binocarlos** [comment]
This needs some work in the `deploy.py` phase - namely to resolve #2

File: install.py [hunk 1]
```diff
@@ -24,6 +24,24 @@
 yum install -y clusterhq-flocker-node
 """)
 
+    # if the dataset.backend is ZFS then install ZFS and mount a flocker pool
+    if c.config["agent_config"]["dataset"]["backend"] == "zfs":
+        # CentOS ZFS installation requires a restart
+        # XXX todo - find out a way to handle a restart mid-script
+        if c.config["os"] == "centos":
+            print "Auto-install of ZFS on centos not currently supported - restart required"
```
Comment (lukemarsden) [inline install.py]: maybe `sys.exit(1)` here

File: install.py [hunk 2]
```diff
@@ -24,6 +24,24 @@
...
+
+        for node in c.config["agent_nodes"]:
+            if c.config["os"] == "ubuntu":
+                c.runSSH(node, """echo installing-zfs
+add-apt-repository -y ppa:zfs-native/stable
+apt-get update
+apt-get -y --force-yes install libc6-dev zfsutils
+mkdir -p /var/opt/flocker
+truncate --size 10G /var/opt/flocker/pool-vdev || true
+zpool create flocker /var/opt/flocker/pool-vdev || true
```
Comment (lukemarsden) [inline install.py]: why these `|| true`? surely it's a problem if this fails and we want to fail?

**Comment by lukemarsden** [comment]
Thanks, in general looks like a good incremental improvement towards getting the zfs backend working. It would be nice to have end-to-end test coverage of all this stuff, but we can do that as a separate issue in the labs JIRA. Please address the review comments and resubmit.

**Comment by binocarlos** [comment]
Thanks! I've addressed those comments - there is an end-to-end test coming together that will end up in this repo :-)

**Comment by binocarlos** [comment]
@lukemarsden thanks for the tip - looking at the old version it's obvious there would be a bunch of spaces in the output! (dups)

**Comment by robhaswell** [comment]
This branch is going to have been hosed by the setuptools changes.

**Comment by lukemarsden** [comment]
@binocarlos could you merge master now that setuptools changes have landed? And maybe fix key distribution for zfs while you're at it?

Copying https://github.com/ClusterHQ/powerstrip-flocker/commit/0eb10025b42dc52a4453d799f552d83e8dec8543 would probably suffice for this.

This would allow us to close #2 :smile:

**Comment by lukemarsden** [comment]
This PR also needs associated docs changes.

**Comment by binocarlos** [comment]
Am closing this and will re-open a PR against the most recent master that also has SSH key distribution plugged in

**Comment by binocarlos** [comment]
OK I opened #24 instead

**Pull request closed without merging**